judge/views/problem.py

```diff
@@ -684,6 +732,46 @@ def form_valid(self, form):
         return HttpResponseRedirect(reverse('admin:judge_problem_change', args=(problem.id,)))
 
 
+class ProblemSuggest(TitleMixin, CreateView):
+    template_name = 'problem/suggest.html'
+    model = Problem
+    form_class = ProblemSuggestForm
+
+    def get_title(self):
+        return _('Suggesting new problem')
+
+    def get_content_title(self):
+        return _('Suggesting new problem')
+
+    def post(self, request, *args, **kwargs):
+        form = ProblemSuggestForm(request.POST or None)
+        if form.is_valid():
+            print("valid vcl")
+            problem = form.save()
+            problem.is_suggesting = True
+            problem.curators.add(request.user.profile)
+            problem.save()
+            return self.form_valid(form)
+        else:
+            print("form gay")
+            return self.form_invalid(form)
+
+    def dispatch(self, request, *args, **kwargs):
+        try:
+            if request.user.has_perm('judge.edit_own_problem'):
```

Comment:
hmm, should we create new permission like `can_suggest_problem`?

Reply:
Yeah sure :)